src/components/Sidebar.js

```diff
@@ -1,13 +1,72 @@
-import { Link } from 'react-router-dom';
+import { useState } from 'react';
+import { Link, useLocation } from 'react-router-dom';
+import { Icon } from '@iconify/react';
 
 export default function Sidebar() {
+  const [open, setOpen] = useState(true);
+
   return (
-    <div className="w-60 p-4 flex-shrink-0 shadow-xl">
-      <h1 className="font-semibold text-2xl mb-8">Coverage Checker</h1>
-      <nav className="flex flex-col gap-2">
-        <Link to="/file-upload">Upload Files</Link>
-        <Link to="/">mCODE Coverage</Link>
-      </nav>
+    <div className={`flex flex-col ${open ? 'w-60' : 'w-20'} p-4 h-screen sticky top-0 flex-shrink-0 shadow-xl`}>
```

Comment:
We could edit some other styles here: 
- Let's change the shadow to be more intense, using the `shadow-sidebarButton` class
- Let's add the transition styles to this, using the `transition-all` class

Reply:
Also, let's change the collapsed width styling to `w-[4.5rem]`, which helps with the centering of the selected icon w/r/t the box shadow.